src/main/java/dk/cngroup/lentils/controller/ClientController.java

```diff
@@ -0,0 +1,101 @@
+    private static final String REDIRECT_CLIENT_VIEW_CYPHER_LIST = "redirect:/client/list";
+    private static final String REDIRECT_CLIENT_VIEW_CYPHER_DESCRIPTION = "redirect:/client/detail";
+
+    private CypherService cypherService;
+    private TeamService teamService;
+    private StatusService statusService;
+    private HintService hintService;
+    private HintTakenService hintTakenService;
+
+    @Autowired
+    public ClientController(final CypherService cypherService,
+                            final TeamService teamService,
+                            final StatusService statusService,
+                            final HintService hintService,
+                            final HintTakenService hintTakenService) {
+        this.cypherService = cypherService;
+        this.teamService = teamService;
+        this.statusService = statusService;
+        this.hintService = hintService;
+        this.hintTakenService = hintTakenService;
+    }
+
+    @GetMapping(value = "/list")
+    public String listAllCyphers(final Model model) {
+        model.addAttribute("cyphers", cypherService.getAll());
+        model.addAttribute("team", teamService.getTeam(2L));
+        model.addAttribute("hints", hintService.getAll());
+        model.addAttribute("status", CypherStatus.SKIPPED);
+        return CLIENT_VIEW_CYPHER_LIST;
+    }
+
+    @GetMapping(value = "/detail/{id}")
+    public String cypherDetail(@PathVariable("id") final Long id, final Model model) {
+        Cypher cypher = cypherService.getCypher(id);
+        String status = statusService.getStatusName(teamService.getTeam(2L), cypher);
+        model.addAttribute("team", teamService.getTeam(2L));
+        model.addAttribute("cypher", cypher);
+        model.addAttribute("statusName", status);
+        model.addAttribute("hintsNotTaken",
+                hintService.getHintsNotTakenByTeam(cypher,
+                        teamService.getTeam(2L)));
+        model.addAttribute("hintsTaken",
+                hintTakenService.getAllByTeamAndCypher(teamService.getTeam(2L), cypher));
+        model.addAttribute("nextCypher", cypherService.getNext(cypher.getStage()));
+        return CLIENT_VIEW_CYPHER_DESCRIPTION;
+    }
+
+    @PostMapping(value = "/detail/verify")
+    public String verifyCodeword(final Cypher cypher, final Model model) {
+        model.addAttribute("cypher", cypher);
+        if (cypherService.checkCodeword(cypher.getCodeword(), cypher.getCypherId())) {
+            statusService.markCypher(cypherService.getCypher(cypher.getCypherId()),
+                    teamService.getTeam(2L),
+                    CypherStatus.SOLVED);
+        }
+        return REDIRECT_CLIENT_VIEW_CYPHER_DESCRIPTION + "/" + cypher.getCypherId();
+    }
+
+    @RequestMapping(value = "/detail/takeHint/{hintId}")
+    public String getHint(@PathVariable("hintId") final Long id, final Cypher cypher, final Model model) {
```

Comment:
Nejsem si jisty, jestli toto ma byt get metoda. Zkontroluj to prosim v Progress kontroleru, co implementoval David. At je to konzistentni.